src/resnext.jl

```diff
@@ -0,0 +1,137 @@
+- `outplanes`: a list of the number of output feature maps for each convolution
+               within the residual block
+- `cardinality`: the number of groups to use for the convolution
+- `downsample`: set to `true` to downsample the input
+"""
+resnextblock(inplanes, outplanes, cardinality, downsample = false) = downsample ? 
+    Chain(
+        conv_bn((1, 1), inplanes, outplanes[1]; stride = 1, bias = false)...,
+        conv_bn((3, 3), outplanes[1], outplanes[2]; stride = 2, pad = 1, 
+                    groups = cardinality, bias = false)...,
+        conv_bn((1, 1), outplanes[2], outplanes[3], identity; stride = 1, bias = false)...) :
+    Chain(
+        conv_bn((1, 1), inplanes, outplanes[1]; stride = 1, bias = false)...,
+        conv_bn((3, 3), outplanes[1], outplanes[2]; stride = 1, pad = 1,
+                    groups = cardinality, bias = false)...,
+        conv_bn((1, 1), outplanes[2], outplanes[3], identity; stride = 1, bias = false)...)
+
+"""
+    resnext(connection = (x, y) -> @. relu(x) + relu(y); channel_config, block_config, 
+            cardinality, nclasses = 1000)
+    
+Create a ResNeXt model
+([reference](https://arxiv.org/abs/1611.05431)).
+
+# Arguments
+- `connection`: the binary function applied to the output of residual and skip paths in a block
+- `channel_config`: the growth rate of the output feature maps within a residual block
+- `block_config`: a list of the number of residual blocks at each stage
+- `cardinality`: the number of groups to use for the convolution
+- `nclasses`: the number of output classes
+"""
+function resnext(connection = (x, y) -> @. relu(x) + relu(y); channel_config, block_config, cardinality, nclasses = 1000)
+    inplanes = 128
+    baseplanes = 128
+    layers = []
+    append!(layers, conv_bn((7, 7), 3, inplanes; stride = 2, pad = (3, 3)))
+    push!(layers, MaxPool((3, 3), stride = (2, 2), pad = (1, 1)))
+    for (i, nrepeats) in enumerate(block_config)
+        # output planes within a block
+        outplanes = baseplanes .* channel_config
+        # push first skip connection on using first residual
+        # downsample the residual path if this is the first repetition of a block
+        push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, i != 1),
+                                            skip_projection(inplanes, outplanes[end], i != 1)))
+        # push remaining skip connections on using second residual
+        inplanes = outplanes[end]
+        for _ in 2:nrepeats
+            push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, false),
+                                                skip_identity(inplanes, outplanes[end], false)))
+            inplanes = outplanes[end]
+        end
+        # next set of output plane base is doubled
+        baseplanes *= 2
+    end
+
+    return Chain(Chain(layers...),
+                Chain(AdaptiveMeanPool((1, 1)), flatten, Dense(inplanes, nclasses)))
+end
```

Comment:
As a result of my comment about `resnextblock`, we don't really need `channel_config` here. This is one of the features of ResNeXt...it simplifies the design quite a bit by thinking more generically. In the paper, `channel_config == [1, 1, 2]` always, so we could just hardcode the fact that the output channels doubles at the end of the block. But I think future-proof this a bit by adding a `channel_multiplier`. Here's how I would do it.

Also, note a bug that `inplanes = 64` to start.

```suggestion
function resnext(cardinality, width, channel_multiplier = 2, connection = (x, y) -> @. relu(x) + relu(y);
                 block_config, nclasses = 1000)
    inplanes = 64
    baseplanes = 128
    layers = []
    append!(layers, conv_bn((7, 7), 3, inplanes; stride = 2, pad = (3, 3)))
    push!(layers, MaxPool((3, 3), stride = (2, 2), pad = (1, 1)))
    for (i, nrepeats) in enumerate(block_config)
        # output planes within a block
        outplanes = baseplanes * channel_multiplier
        # push first skip connection on using first residual
        # downsample the residual path if this is the first repetition of a block
        push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, width, i != 1),
                                           skip_projection(inplanes, outplanes, i != 1)))
        # push remaining skip connections on using second residual
        inplanes = outplanes
        for _ in 2:nrepeats
            push!(layers, Parallel(connection, resnextblock(inplanes, outplanes, cardinality, width, false),
                                               skip_identity(inplanes, outplanes, false)))
        end
        baseplanes = outplanes
    end

    return Chain(Chain(layers...),
                 Chain(AdaptiveMeanPool((1, 1)), flatten, Dense(inplanes, nclasses)))
end
```

Reply:
I'd rather avoid hard coding it if there are variants of interest that diverge from the default. It keeps the design simpler by hoisting the `channel_multiplier` outside the package. If the dimensions don't align, that would error on the forward pass, so we should be fine to report those.

Reply:
Not sure what you mean exactly, but my suggestion is not to hardcode it.